src/main/java/org/candlepin/async/tasks/CertificateCleanupJob.java

```diff
@@ -0,0 +1,132 @@
+    private final IdentityCertificateCurator identCertCurator;
+    private final ContentAccessCertificateCurator caCertCurator;
+    private final CertificateSerialCurator serialCurator;
+
+    @Inject
+    public CertificateCleanupJob(
+        Configuration config,
+        ConsumerCurator consumers,
+        IdentityCertificateCurator identCerts,
+        ContentAccessCertificateCurator contentAccessCerts,
+        CertificateSerialCurator serialCurator) {
+        this.config = Objects.requireNonNull(config);
+        this.consumerCurator = Objects.requireNonNull(consumers);
+        this.identCertCurator = Objects.requireNonNull(identCerts);
+        this.caCertCurator = Objects.requireNonNull(contentAccessCerts);
+        this.serialCurator = Objects.requireNonNull(serialCurator);
+    }
+
+    @Override
+    @Transactional
+    public void execute(JobExecutionContext context) throws JobExecutionException {
+        log.debug("Starting certificate cleanup.");
+        // Skipping entitlement certificates as those are handled by ExpiredPoolsCleanupJob
+        cleanupExpiredIdentityCerts();
+        cleanupExpiredContentAccessCerts();
+        cleanupCertificateSerials();
+        log.debug("Certificate cleanup successfully completed.");
+    }
+
+    private void cleanupExpiredIdentityCerts() {
+        List<ExpiredCertificate> allExpiredIdCertificates = this.identCertCurator.listAllExpired();
+        log.info("Cleaning up {} expired identity certificates.", allExpiredIdCertificates.size());
+        for (List<ExpiredCertificate> expiredCertificates : partition(allExpiredIdCertificates)) {
+            List<String> expiredCertIds = certIdsOf(expiredCertificates);
+            List<Long> expiredSerials = serialsOf(expiredCertificates);
+            List<String> consumersToUnlink = this.consumerCurator.findUuidsByIdentCerts(expiredCertIds);
+            int unlinkedConsumers = this.consumerCurator.unlinkIdCertificatesOf(consumersToUnlink);
+            log.debug("Unlinking identity certificates of {} consumers.", unlinkedConsumers);
+            this.identCertCurator.deleteByIds(expiredCertIds);
+            this.serialCurator.revokeByIds(expiredSerials);
+        }
+    }
+
+    private void cleanupExpiredContentAccessCerts() {
+        List<ExpiredCertificate> allExpiredCaCertificates = this.caCertCurator.listAllExpired();
+        log.info("Cleaning up {} expired content access certificates.", allExpiredCaCertificates.size());
+        for (List<ExpiredCertificate> expiredCertificates : partition(allExpiredCaCertificates)) {
+            List<String> expiredCertIds = certIdsOf(expiredCertificates);
+            List<Long> expiredSerials = serialsOf(expiredCertificates);
+            List<String> consumersToUnlink = this.consumerCurator
+                .findUuidsByContentAccessCerts(expiredCertIds);
+            int unlinkedConsumers = this.consumerCurator.unlinkCaCertificatesOf(consumersToUnlink);
```

Comment:
Same here: a single query should be used.

Reply:
Done.